apps/CommonApp/src/ListWithHeader/ListWithHeader.tsx

```diff
@@ -97,46 +97,50 @@ interface ScrollComponentWithOffsetProps extends ScrollViewProps {
   scrollOffset: SharedValue<number>;
   animatedScrollEnabled: SharedValue<boolean>;
   dragGesture: GestureType;
+  ref?: React.RefObject<Animated.ScrollView | null>;
 }
 
-const ScrollComponentWithOffset = React.forwardRef(
-  (props: ScrollComponentWithOffsetProps, ref: any) => {
-    const scrollRef = useAnimatedRef<Animated.ScrollView>();
-    const scrollViewOffset = useScrollViewOffset(scrollRef);
-
-    useAnimatedReaction(
-      () => {
-        return scrollViewOffset.value;
-      },
-      (offset) => {
-        props.scrollOffset.value = offset;
-      }
-    );
+const ScrollComponentWithOffset = ({
+  ref,
+  ...props
+}: ScrollComponentWithOffsetProps) => {
+  const scrollRef = useAnimatedRef<Animated.ScrollView>();
+  const scrollViewOffset = useScrollViewOffset(scrollRef);
+
+  useAnimatedReaction(
+    () => {
+      return scrollViewOffset.value;
+    },
+    (offset) => {
+      props.scrollOffset.value = offset;
+    }
+  );
 
-    useEffect(() => {
+  useEffect(() => {
+    if (ref) {
       ref.current = scrollRef.current;
```

Comment:
I don't think this ref is being cleaned up at all, but it's hard to judge how much it matters in the internal test app.